### PR TITLE
github/ci: Dont trigger command workflow for bots

### PR DESCRIPTION
**Pull request by phlax**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .github/workflows/commands.yml [hunk 1]
```diff
@@ -5,7 +5,13 @@ on:
 
 jobs:
   retest:
-    if: github.repository == 'envoyproxy/envoy'
+    if: |
+      ${{
+         github.event.issue.pull_request
+         && github.repository == 'envoyproxy/envoy'
+         && github.actor != 'repokitteh-read-only[bot]'
+         && github.actor != 'dependabot[bot]'
```
Comment (phlax) [inline .github/workflows/commands.yml:13]: i think these conditions should work correctly - but not 100% - would be good to exclude all bots

Comment (wbpcode) [reply .github/workflows/commands.yml:13]: because AND is use here, so seems this new condition is more strict?

**Comment by phlax** [comment]
for ref its reducing the footprint here https://github.com/envoyproxy/envoy/actions/workflows/commands.yml

there is already some contention (wait to see how long before this comment pops up ...)

im starting to move some jobs out of azp - partly to reduce contention there and partly because that is the plan (cf #27294 ) so the more we can prevent waste on the machines the better

**Comment by phlax** [comment]
relatedly this probably would be better to do with rk cc @itayd

**Comment by phlax** [comment]
@wbpcode im gonna push this through - the ci passed but is messed up here not sure why - i think unrelated

i do want to test this on postsubmit tho - so will watch/fix etc now

**Comment by phlax** [comment]
my first question is what happens on closed prs

/retest

**Comment by phlax** [comment]
i think both triggered - altho only rk posted here - probably we want to add the pr state to the condition

**Comment by phlax** [comment]
hmm - this didnt work exactly - its probably made it faster - it only skips the job now but still sets itself up

**Pull request merged**